File: 2025/Company/Company.Web/Modules/ServerTypes/Company/ProvinceForm.ts
import { StringEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { CountryDialog } from "../../Company/Country/CountryDialog";

export interface ProvinceForm {
    ProvinceCode: StringEditor;
    CountryId: LookupEditor;
    ProvinceName: StringEditor;
}

export class ProvinceForm extends PrefixedContext {
    static readonly formKey = 'Company.Province';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProvinceForm.init)  {
            ProvinceForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(ProvinceForm, [
                'ProvinceCode', w0,
                'CountryId', w1,
                'ProvinceName', w0
            ]);
        }
    }
}

queueMicrotask(() => [CountryDialog]); // referenced dialogs

File: 2025/Company/Company.Web/Modules/Company/Province/ProvinceForm.cs
namespace Company.Company.Forms;

[FormScript("Company.Province")]
[BasedOnRow(typeof(ProvinceRow), CheckNames = true)]
public class ProvinceForm
{
    public string ProvinceCode { get; set; }
    public int CountryId { get; set; }
    public string ProvinceName { get; set; }
}